File: TheProgrammeOfArt/DemoOfChapter2/IPCDemo/app/src/main/java/com/example/ipcdemo/SecondActivity.java
package com.example.ipcdemo;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.ipcdemo.entity.User;
import com.example.ipcdemo.utils.IPCByFile;
import com.example.ipcdemo.utils.MyConstants;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);


        findViewById(R.id.btn_activity_second).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                IPCByFile.recoverFromFile();
            }
        });


    }


}

File: LittleDemo/JetpackDemo/workmanager/src/main/java/com/example/workmanager/SecondActivity.kt
package com.example.workmanager

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.JsonReader
import android.util.Log
import androidx.work.*
import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import com.alibaba.fastjson.JSONPObject
import com.example.workmanager.databinding.ActivityMainBinding
import com.example.workmanager.work.*
import java.util.concurrent.TimeUnit

class SecondActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.button.setOnClickListener {
            addWork()
        }


    }



    private fun addWork() {

        val aWork = OneTimeWorkRequest
            .Builder(AWorker::class.java)
            .build()

        val bWork = OneTimeWorkRequest
            .Builder(BWorker::class.java)
            .build()

        val cWork = OneTimeWorkRequest
            .Builder(CWorker::class.java)
            .build()

        val dWork = OneTimeWorkRequest
            .Builder(DWorker::class.java)
            .build()

        val eWork = OneTimeWorkRequest
            .Builder(EWorker::class.java)
            .build()

        val workContinuation1 = WorkManager.getInstance(this)
            .beginWith(aWork)
            .then(bWork)


        val workContinuation2 = WorkManager.getInstance(this)
            .beginWith(cWork)
            .then(dWork)

        val taskList = mutableListOf<WorkContinuation>()
        taskList.add(workContinuation1)
        taskList.add(workContinuation2)

        //任务组合
        WorkContinuation.combine(taskList)
            .then(eWork)
            .enqueue()

//        val workManager = WorkManager.getInstance(this)
//        //任务提交给WorkManager
//        workManager
//            .beginWith(aWork)
//            .then(bWork)
//            .enqueue()
    }
}